### PR TITLE
adds an e2e load test scenario for encryption/migration

### PR DESCRIPTION
**Pull request by p0lyn0mial**
the following scenario will be used by https://github.com/openshift/cluster-openshift-apiserver-operator/pull/261 and https://github.com/openshift/cluster-kube-apiserver-operator/pull/654

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
proof that it actually works https://github.com/openshift/cluster-kube-apiserver-operator/pull/654 and https://github.com/openshift/cluster-openshift-apiserver-operator/pull/261

File: test/library/encryption/perf_helpers.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+			wg.Add(1)
+			go work(wg)
+		}
+		wg.Wait()
+		return nil
+	}
+}
+
+func createNamespaceName() string {
+	return fmt.Sprintf("encryption-%s", rand.String(10))
```
Comment (sttts) [inline test/library/encryption/perf_helpers.go]: is this random enough?

File: test/library/encryption/perf_scenarios.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+type GetOperatorConditionsFuncType func(t testing.TB) ([]v1.OperatorCondition, error)
+
+type PerfScenario struct {
+	BasicScenario
+	GetOperatorConditionsFunc GetOperatorConditionsFuncType
+
+	DBLoaderFunc          DBLoaderFuncType
+	AssertDBPopulatedFunc func(t testing.TB, errorStore map[string]int, statStore map[string]int)
+	AssertMigrationTime   func(t testing.TB, migrationTime time.Duration)
+	// DBLoaderWorker the number of workers that will execute DBLoaderFunc
```
Comment (sttts) [inline test/library/encryption/perf_scenarios.go]: nit: is the number

File: test/library/encryption/perf_helpers.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+func DBLoaderRepeat(times int, genNamespaceName bool, workToRepeatFunc ...DBLoaderFuncType) DBLoaderFuncType {
+	return DBLoaderRepeatParallel(times, 1, genNamespaceName, workToRepeatFunc...)
+}
+
+func DBLoaderRepeatParallel(times int, workers int, genNamespaceName bool, workToRepeatFunc ...DBLoaderFuncType) DBLoaderFuncType {
+	return func(kubeClient kubernetes.Interface, namespace string, errorCollector func(error), statsCollector func(string)) error {
+		if times < workers {
+			panic("DBLoaderRepeat cannot be < workers")
+		}
+		wg := &sync.WaitGroup{}
```
Comment (sttts) [inline test/library/encryption/perf_helpers.go]: nit: is there a need for a pointer?

File: test/library/encryption/perf_helpers.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+		workPerWorker := times / workers
+		for w := 0; w < workers; w++ {
+			work := func(wg *sync.WaitGroup) {
+				defer wg.Done()
+				for i := 0; i < workPerWorker; i++ {
+					if genNamespaceName {
+						namespace = createNamespaceName()
+					}
+					err := runWorkFunctions(kubeClient, namespace, errorCollector, statsCollector, workToRepeatFunc...)
```
Comment (sttts) [inline test/library/encryption/perf_helpers.go]: save lines: `if err := ...; err != nil {`

File: test/library/encryption/perf_helpers.go (hunk 1 above)
Comment (p0lyn0mial) [reply test/library/encryption/perf_helpers.go]: yes, the tests we have atm generate `3` namespaces at most :)

Comment (sttts) [reply test/library/encryption/perf_helpers.go]: Yeah, but the chance that two of the 3 hit the same namespace is high https://www.dcode.fr/birthday-problem

**Comment by sttts** [comment]
/approve
/lgtm

/hold
https://github.com/openshift/library-go/pull/599/files#r350215005 is still open.

File: test/library/encryption/perf_helpers.go (hunk 1 above)
Comment (p0lyn0mial) [reply test/library/encryption/perf_helpers.go]: Does it mean we have a 50% chance of a collision after generating `60466176` namespaces?

**Pull request merged**